static/src/javascripts/projects/common/modules/discussion/comment-box.js

```diff
@@ -67,11 +67,17 @@ CommentBox.prototype.classes = {};
  * @override
  */
 CommentBox.prototype.errorMessages = {
+    /* Discussion API error codes - See DAPI exceptions and associated error messages  */
     EMPTY_COMMENT_BODY: 'Please write a comment.',
     COMMENT_TOO_LONG: 'Your comment must be fewer than 5000 characters long.',
-    HTTP_420: 'You can only post one comment every minute. Please try again in a moment.',
-    HTTP_0: /*CORS blocked by HTTP/1.0 proxy*/'Could not post due to your internet settings, which might be controlled by your provider. Please contact your administrator or disable any proxy servers or VPNs and try again.',
     USER_BANNED: 'Commenting has been disabled for this account (<a href="/community-faqs#321a">why?</a>).',
+    DISCUSSION_CLOSED: 'Sorry your comment can not be published as the discussion is now closed for comments.',
+    COMMENT_RATE_LIMIT_EXCEEDED: 'You can only post one comment every minute. Please try again in a moment.',
+    INVALID_PROTOCOL: 'Sorry your comment can not be published as it was not sent over a secure channel. Please report us this issue',
+    AUTH_COOKIE_INVALID: 'Sorry your comment can not be published as your authenciation cookie seems invalid. Please try to sign in before posting again.',
+    'READ-ONLY-MODE': 'Sorry your comment can not currently be published as commenting is undergoing maintenance but will be back shortly. Please try again in a moment.',
+    /* Custom error codes */
+    API_CORS_BLOCKED: /*CORS blocked by HTTP/1.0 proxy*/'Could not post due to your internet settings, which might be controlled by your provider. Please contact your administrator or disable any proxy servers or VPNs and try again.',
```

Comment:
actually discussion uses https now so this can't happen any more, although I suppose some other CORS failure could cause this code